File: qt6_ros_cpp/src/increment_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <string>

int main(int argc, char* argv[]) {
    ros::init(argc, argv, "increment_publisher");
    ros::NodeHandle nh;
    ros::NodeHandle pnh("~");

    std::string chatter_topic;
    pnh.param<std::string>("chatter_topic", chatter_topic, "chatter");

    ros::Publisher pub = nh.advertise<std_msgs::String>(chatter_topic, 10);
    ros::Rate rate(1);

    int count = 0;
    while (ros::ok()) {
        std_msgs::String msg;
        msg.data = "Count: " + std::to_string(count);
        ROS_INFO("%s", msg.data.c_str());
        pub.publish(msg);
        ++count;
        rate.sleep();
    }

    return 0;
}

File: qt6_ros/src/qt6_ros/increment_publisher.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String


def main():
    rospy.init_node('increment_publisher', anonymous=False)
    chatter_topic = rospy.get_param('~chatter_topic', 'chatter')
    pub = rospy.Publisher(chatter_topic, String, queue_size=10)
    rate = rospy.Rate(1)
    count = 0

    while not rospy.is_shutdown():
        message = f"Count: {count}"
        rospy.loginfo(message)
        pub.publish(String(data=message))
        count += 1
        rate.sleep()


if __name__ == '__main__':
    try:
        main()
    except rospy.ROSInterruptException:
        pass
